server/src/main/java/io/crate/planner/operators/Collect.java

```diff
@@ -155,7 +156,8 @@ public ExecutionPlan build(PlannerContext plannerContext,
                 order.map(binder)
                     // Filter out literal constants as ordering by constants is a NO-OP and also not supported
                     // on the collect operation.
-                    .exclude(s -> s instanceof Literal));
+                    .exclude(s -> s instanceof Literal ||
+                                  s instanceof AliasSymbol && ((AliasSymbol)s).symbol() instanceof Literal));
```

Comment:
Could use the new instanceof pattern match syntax:

```suggestion
                                  s instanceof AliasSymbol alias && alias.symbol() instanceof Literal));
```

Reply:
Thank you, I will copy that in.